### PR TITLE
Avoid domain errors in computing TPC wire angles.

### PR DESCRIPTION
**Pull request by PetrilloAtWork**
This solves [issue #25559](https://cdcvs.fnal.gov/redmine/issues/25559).

### CODE REVIEW

**Comment by knoepfel** [comment]
trigger build

File: larcorealg/Geometry/WireGeo.cxx [hunk 1]
```diff
@@ -49,7 +51,7 @@ namespace geo{
     lp.SetZ(fHalfL);
     auto end = toWorldCoords(lp);
 
-    fThetaZ = std::acos((end.Z() - fCenter.Z())/fHalfL);
+    fThetaZ = std::acos(std::clamp((end.Z() - fCenter.Z())/fHalfL, -1.0, +1.0));
```
Comment (absolution1) [inline larcorealg/Geometry/WireGeo.cxx:54]: I guess that one unfortunate byproduct of this is that the clamp would mask genuine problems (e.g. a bug causing the argument to genuinely evaluate to something much larger than 1). But, I guess it's not really the job of this function to catch such bugs...

**Comment by absolution1** [comment]
approve

**Comment by lgarren** [comment]
approve

**Pull request merged**